src/util/ajax.js

```diff
@@ -39,7 +39,29 @@ if (typeof Object.freeze == 'function') {
  * @typedef {Object} RequestParameters
  * @property {string} url The URL to be requested.
  * @property {Object} headers The headers to be sent with the request.
+ * @property {string} method Request method `'GET' | 'POST' | 'PUT'`.
+ * @property {string} body Request body.
+ * @property {string} type Response body type to be returned `'string' | 'json' | 'arrayBuffer'`.
  * @property {string} credentials `'same-origin'|'include'` Use 'include' to send cookies with cross-origin requests.
+ * @property {boolean} collectResourceTiming If true, Resource Timing API information will be collected for these transformed requests and returned in a resourceTiming property of relevant data events.
+ * @example
+ * // transformRequest used to modify requests that begin with `http://myHost`
+ * transformRequest: (url, resourceType)=> {
+ *  if(resourceType === 'Source' && url.startsWith('http://myHost')) {
+ *    return {
+ *      url: url.replace('http', 'https'),
+ *      headers: { 'my-custom-header': true},
+ *      credentials: 'include'  // Include cookies for cross-origin requests
+ *    }
+ *   }
+ *  }
+ * // Example of `RequestParameters` object returned from the above transformRequest function.
+ * {
```

Comment:
Should this be commented out? Or wrapped in `console.log`? It's also missing commas after each key/value.

Reply:
@katydecorah do you think the return object would be better in a separate `@example` block? We could add a caption and it might be clearer what it's trying to convey.

Reply:
My vote would be to remove it as Asheem recommends